### DISCUSSION TITLE
GetPixel().ToColor() does not return correct color after a few runs

### DISCUSSION THREAD

**Discussion by fvds72** [dlemstra/Magick.NET#824] [Help]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of Magick.NET
- [x] I have searched [open](https://github.com/dlemstra/Magick.NET/issues) and [closed](https://github.com/dlemstra/Magick.NET/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
I am trying to trim the left and right side of the image, by checking whether the column has only white pixels, if so than crop that column. The following piece of code does not work when calling multiple times for different images. I see that the function works with the first image I am trying to trim, but the second image I am trying to process l am always get the color white back and will cause the image to be trimmed to size 0,0:

        static void TrimHorizontal(MagickImage theImage)
        {
            bool edgeFound = false;
            int column = 0;
            // Start at trimming Left-side
            while (edgeFound == false)
            {
                using (IPixelCollection<byte> pixels = theImage.GetPixels())
                {
                    for (int y = 0; y < theImage.Height; y++)
                    {
                        Console.WriteLine(String.Format("Checking postion ({0},{1}): {2}", column, y, pixels.GetPixel(0, y).ToColor()));
                        // Detect non white pixel in column/y position 0
                        if (pixels.GetPixel(0, y).ToColor().CompareTo(MagickColors.White) != 0)
                        {
                            edgeFound = true;
                        }
                    }
                }

                if (edgeFound == false)
                {
                    // Remove first pixel column on the left
                    theImage.Crop(theImage.Width - 1, theImage.Height, Gravity.East);
                }

                column++;
            }

            // TODO implement trimming right side as well.
           
            return;
}

I have found of a similar bug in this repository (#403), but that one seems to be fixed. Am I doing something wrong or is there a better method  of Trimming -horizontal.

### Steps to Reproduce
Call the method multiple times in one run (on different images).

### System Configuration

- Magick.NET version: 7.22.2.2
- Environment (Operating system, version and so on): Windows 10, Visual Studio 2015
- Additional information:

**Comment by dlemstra** [comment]
I am unable to reproduce your issue with the following code:

```C#
using (var image = new MagickImage(MagickColors.Red, 1, 1))
{
    image.Extent(3, 3, Gravity.Center, MagickColors.White);
    TrimHorizontal(image);
    Console.WriteLine($"{image.Width}x{image.Height}");

    image.Read(MagickColors.Red, 1, 1);
    image.Extent(3, 3, Gravity.Center, MagickColors.White);
    TrimHorizontal(image);
    Console.WriteLine($"{image.Width}x{image.Height}");
}
```

I suspect that the problem is somewhere in your code. Or maybe you are checking an all white image?

*EDIT*: Just noticed that you are not using the column in `pixels.GetPixel(0, y).`?

**Comment by dlemstra** [comment]
The next version of MagickNET will include a new `TrimVertical` and a new `TrimHorizontal` method so you don't need to create one yourself.

**Comment by fvds72** [comment]
Oh super. Will be using that one

Thanks

Op wo 30 dec. 2020 14:13 schreef Dirk Lemstra <notifications@github.com>:

**Reply by dlemstra** [reply]
Decided to add an overload to `Trim` instead where you could do something like this: `image.Trim(Gravity.East, Gravity.West);`

**Comment by fvds72** [comment]
Sounds good, please keep me updated when you release this implementation.

Thanks for the help! (and have a good 2021!)
